crepecake/src/main/java/com/google/cloud/tools/crepecake/tar/TarStreamBuilder.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.tar;
+
+import com.google.cloud.tools.crepecake.blob.Blob;
+import com.google.cloud.tools.crepecake.blob.Blobs;
+import com.google.common.io.ByteStreams;
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+
+/** Builds a tarball archive. */
+public class TarStreamBuilder {
```

Comment:
I wonder if this class should provide a stronger contract to its internals. Since we're delegating to "writeEntriesAsTarArchive" kinda lazily, is it possible that we call addFile after toBlob and change the expectation of what's written in toBlob?

Would it make sense to have a TarBlobBuilder that returns an immutable TarBlob?

Reply:
I changed this to make it be locked-in when calling toBlob.